src/main/java/de/tum/cit/aet/artemis/programming/service/localci/LocalCIResultProcessingService.java

```diff
@@ -264,6 +279,37 @@ public BuildJob saveFinishedBuildJob(BuildJobQueueItem queueItem, BuildStatus bu
         }
     }
 
+    private void updateExerciseBuildDurationAsync(ProgrammingExercise exercise) {
+        CompletableFuture.runAsync(() -> updateExerciseBuildDuration(exercise));
+    }
+
+    private void updateExerciseBuildDuration(ProgrammingExercise exercise) {
+        try {
+            ProgrammingExerciseBuildStatistics buildStatistics = programmingExerciseBuildStatisticsRepository.findByExerciseId(exercise.getId()).orElse(null);
+            long successfulBuildJobCountByExerciseId = buildJobRepository.fetchSuccessfulBuildJobCountByExerciseId(exercise.getId());
+            boolean shouldUpdate = successfulBuildJobCountByExerciseId > 0
+                    && (buildStatistics == null || successfulBuildJobCountByExerciseId - buildStatistics.getBuildCountWhenUpdated() >= 10);
```

Comment:
Why is this 10? What does the 10 mean here? We update every 10 seconds? 
Could we please move it to a constant variable? 

Maybe also make this `(buildStatistics == null || successfulBuildJobCountByExerciseId - buildStatistics.getBuildCountWhenUpdated() >= 10)` a variable with a descriptive name